src/gc/gchandletableimpl.h

```diff
@@ -6,33 +6,49 @@
 #define GCHANDLETABLE_H_
 
 #include "gcinterface.h"
+#include "objecthandle.h"
 
-class GCHandleTable : public IGCHandleTable
+class GCHandleStore : public IGCHandleStore
 {
 public:
-    virtual bool Initialize();
+    GCHandleStore(HandleTableBucket *bucket) 
+        : _underlyingBucket(bucket)
+        { }
 
-    virtual void Shutdown();
+    virtual void Uproot();
 
-    virtual void* GetGlobalHandleStore();
+    virtual bool ContainsHandle(OBJECTHANDLE handle);
 
-    virtual void* CreateHandleStore(void* context);
+    virtual OBJECTHANDLE CreateHandleOfType(Object* object, int type);
 
-    virtual void* GetHandleContext(OBJECTHANDLE handle);
+    virtual OBJECTHANDLE CreateHandleOfType(Object* object, int type, int heapToAffinitizeTo);
 
-    virtual void DestroyHandleStore(void* store);
+    virtual OBJECTHANDLE CreateHandleWithExtraInfo(Object* object, int type, void* pExtraInfo);
 
-    virtual void UprootHandleStore(void* store);
+    virtual OBJECTHANDLE CreateDependentHandle(Object* primary, Object* secondary);
 
-    virtual bool ContainsHandle(void* store, OBJECTHANDLE handle);
+    virtual ~GCHandleStore();
 
-    virtual OBJECTHANDLE CreateHandleOfType(void* store, Object* object, int type);
+private:
+    HandleTableBucket* _underlyingBucket;
```

Comment:
> This is purely GC-side implementation work, so it can be changed independently of the interface work.

That's true, but since you are working this work in the live tree, it would be a good practice to do the refactorings together with the perf mitigations to reduce their perf overhead.